contrib/platform/src/com/sun/jna/platform/win32/Winspool.java

```diff
@@ -565,6 +633,313 @@ boolean OpenPrinter(
+            implements Structure.ByReference {
+        }
+
+        /**
+         * The type to be watched.
+         */
+        public short Type;
+
+        /**
+         * Reserved.
+         */
+        public short Reserved0;
+
+        /**
+         * Reserved.
+         */
+        public int Reserved1;
+
+        /**
+         * Reserved.
+         */
+        public int Reserved2;
+
+        /**
+         * The number of elements in the pFields array.
+         */
+        public int Count;
+
+        /**
+         * A pointer to an array of values. Each element of the array specifies
+         * a job or printer information field of interest.
+         */
+        public Pointer pFields;
+
+        public void setFields(short[] fields) {
+            final long shortSizeInBytes = 2L;
+            Memory fieldsMemory = new Memory(fields.length * shortSizeInBytes);
+            fieldsMemory.write(0, fields, 0, fields.length);
+            pFields = fieldsMemory;
+            Count = fields.length;
+        }
+
+        public short[] getFields() {
+            return pFields.getShortArray(0, Count);
+        }
+    }
+
+    /**
+     * The PRINTER_NOTIFY_INFO structure contains printer information returned
+     * by the FindNextPrinterChangeNotification function. The function returns
+     * this information after a wait operation on a printer change notification
+     * object has been satisfied.
+     *
+     * @see
+     * <a href="https://docs.microsoft.com/en-us/windows/win32/printdocs/printer-notify-info">
+     *     PRINTER_NOTIFY_INFO structure
+     * </a>
+     */
+    @Structure.FieldOrder({ "Version", "Flags", "Count", "aData" })
+    public class PRINTER_NOTIFY_INFO extends Structure {
+
+        /**
+         * The version of this structure. Set this member to 2.
+         */
+        public int Version;
+
+        /**
+         * A bit flag that indicates the state of the notification structure. If
+         * the PRINTER_NOTIFY_INFO_DISCARDED bit is set, it indicates that some
+         * notifications had to be discarded.
+         */
+        public int Flags;
+
+        /**
+         * The number of PRINTER_NOTIFY_INFO_DATA elements in the aData array.
+         */
+        public int Count;
+
+        /**
+         * An array of PRINTER_NOTIFY_INFO_DATA structures. Each element of the
+         * array identifies a single job or printer information field, and
+         * provides the current data for that field.
+         */
+        public PRINTER_NOTIFY_INFO_DATA[] aData =
+            new PRINTER_NOTIFY_INFO_DATA[1];
+
+        @Override
+        public void read() {
+            int count = (Integer) readField("Count");
+            aData = new PRINTER_NOTIFY_INFO_DATA[count];
+            if (count == 0) {
+                Count = count;
+                Version = (Integer) readField("Version");
+                Flags = (Integer) readField("Flags");
```

Comment:
Please consider setting sData to the empty array - then you can always directly iterate over that array.

Reply:
Not sure what you mean - `aData` will become an empty array on line 782.

Reply:
Sorry! Overlooked.